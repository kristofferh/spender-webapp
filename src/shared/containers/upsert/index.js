import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import EditForm from "shared/components/edit-form";

import { upsertItem, fetchItem } from "./actions";

export class Upsert extends Component {
  constructor(props) {
    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentDidMount() {
    if (this.props.id) {
      this.props.fetchItem(this.props.id);
    }
  }

  handleSubmit(values) {
    this.props.upsertItem({ ...values, id: this.props.id });
  }

  render() {
    return (
      <div>
        <h1>{this.props.id ? "Edit" : "Add"}</h1>
        <EditForm
          onSubmit={this.handleSubmit}
          initialValues={this.props.initialValues}
          errors={this.props.errors}
        />
      </div>
    );
  }
}

Upsert.propTypes = {
  upsertItem: PropTypes.func,
  fetchItem: PropTypes.func,
  match: PropTypes.object,
  initialValues: PropTypes.object,
  errors: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  id: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
};

const mapStateToProps = (state, ownProps) => {
  const { upsert: { item: initialValues, errors } } = state;
  const { match: { params: { id = undefined } } } = ownProps;
  return {
    ...state,
    initialValues,
    errors,
    id
  };
};

export default connect(mapStateToProps, { upsertItem, fetchItem })(Upsert);
