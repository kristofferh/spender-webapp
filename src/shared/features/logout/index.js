import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import Cookies from "js-cookie";

import { userLogout } from "shared/data/user/actions";

export class Logout extends Component {
  static propTypes = {
    userLogout: PropTypes.func,
    history: PropTypes.object,
  };

  componentDidMount() {
    Cookies.remove(SESSION_COOKIE);
    this.props.userLogout();
    this.props.history.push("/");
  }

  render() {
    return <div />;
  }
}

export default connect(null, { userLogout })(Logout);
