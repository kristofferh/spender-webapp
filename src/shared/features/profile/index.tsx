import { withFormik } from "formik";
import React from "react";
import ProfileForm, {
  Props as ProfileFormProps
} from "../../components/profile-form";
import { Container, Title } from "./styles";

export type FormValues = {
  avatar?: string;
  firstName?: string;
};

type FormProps = {
  initialValues: FormValues;
  onSubmit: (values: FormValues) => Promise<any>;
};

const ProfileWrapper = withFormik<FormProps & ProfileFormProps, FormValues>({
  enableReinitialize: true,
  mapPropsToValues: ({ initialValues }) => {
    return initialValues;
  },
  handleSubmit: async (values, { setStatus, props: { onSubmit } }) => {
    try {
      await onSubmit(values);
    } catch {
      setStatus("Something went weird.");
    }
  }
})(ProfileForm);

const Profile: React.FC = () => {
  const handleSubmit = async (values: FormValues) => {
    console.log("submit", values);
    Promise.reject("nope");
  };

  const handleDrop = (files: any) => {
    console.log("handleDrop", files);
  };

  return (
    <Container>
      <Title>Profile</Title>
      <ProfileWrapper
        onDrop={handleDrop}
        onSubmit={handleSubmit}
        initialValues={{ avatar: "", firstName: "" }}
      />
    </Container>
  );
};

export default Profile;
