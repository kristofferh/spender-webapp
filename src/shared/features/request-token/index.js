import React, { Component } from "react";
import { withFormik } from "formik";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import SignupForm from "shared/components/signup-form";

import { requestToken } from "./actions";

const SignupWrapper = withFormik({
  enableReinitialize: true,
  mapPropsToValues: () => ({
    email: ""
  }),
  handleSubmit: (values, { props: { onSubmit } }) => {
    onSubmit(values);
  }
})(SignupForm);

export class RequestToken extends Component {
  static propTypes = {
    requestToken: PropTypes.func,
    success: PropTypes.bool,
    isRequesting: PropTypes.bool,
    errors: PropTypes.array
  };

  handleSubmit = values => {
    this.props.requestToken({ delivery: values.email, deliveryType: "email" });
  };

  render() {
    return <SignupWrapper {...this.props} onSubmit={this.handleSubmit} />;
  }
}

const mapStateToProps = state => {
  const { requestToken } = state;
  return { ...requestToken };
};

export default connect(
  mapStateToProps,
  { requestToken }
)(RequestToken);
