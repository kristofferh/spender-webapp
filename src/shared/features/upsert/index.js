import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { reduxForm, SubmissionError } from "redux-form";

import EditForm from "shared/components/edit-form";

import { upsertItem, fetchItem, fetchTags } from "./actions";

const EditWrapper = reduxForm({
  form: "Edit",
  enableReinitialize: true
})(EditForm);

export class Upsert extends Component {
  static propTypes = {
    upsertItem: PropTypes.func,
    fetchItem: PropTypes.func,
    fetchTags: PropTypes.func,
    match: PropTypes.object,
    initialValues: PropTypes.object,
    errors: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
    id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
    tags: PropTypes.array,
    history: PropTypes.object
  };

  static defaultProps = {
    initialValues: {},
    tags: []
  };

  componentDidMount() {
    if (this.props.id) {
      this.props.fetchItem(this.props.id);
    }
    this.props.fetchTags();
  }

  handleSubmit = values => {
    return this.props
      .upsertItem({ ...values, id: this.props.id })
      .then(() => {
        this.props.history.push("/items");
      })
      .catch(errors => {
        console.log("something went weird", errors); // eslint-disable-line no-console
        throw new SubmissionError({ _error: "Something went weird." });
      });
  };

  render() {
    return (
      <div>
        <h1>{this.props.id ? "Edit" : "Add"}</h1>
        <EditWrapper
          onSubmit={this.handleSubmit}
          initialValues={this.props.initialValues}
          errors={this.props.errors}
          tags={this.props.tags}
        />
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  const {
    upsert: { tags, item: initialValues, errors }
  } = state;
  const {
    match: {
      params: { id = undefined }
    }
  } = ownProps;
  return {
    ...state,
    tags,
    initialValues: id ? initialValues : {},
    errors,
    id
  };
};

export default connect(mapStateToProps, { upsertItem, fetchItem, fetchTags })(
  Upsert
);
