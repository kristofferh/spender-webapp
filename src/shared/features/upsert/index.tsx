import { withFormik } from "formik";
import moment from "moment";
import React, { Component } from "react";
import { connect } from "react-redux";
import EditForm, { Props as EditFormProps } from "shared/components/edit-form";
import { deleteItem, fetchItem, upsertItem } from "shared/data/item/actions";
import { omit } from "shared/utils/object";
import { Container, Title } from "./styles";

export type Tag = {
  name: string;
  color?: string;
};

export type FormValues = {
  date?: string;
  amount: string | number;
  description: string;
  tags: Tag[];
};

type Props = {
  upsertItem: ({}) => Promise<any>;
  fetchItem: (id: string | number) => void;
  fetchTags: () => void;
  deleteItem: (id: string | number) => Promise<any>;
  match: {};
  initialValues: FormValues;
  errors: {};
  id: number | string;
  tags: Tag[];
  history: {
    push: (path: string) => void;
  };
  isSubmitting?: boolean;
};

type FormProps = {
  initialValues: FormValues;
  onSubmit: (values: FormValues) => Promise<any>;
};

const EditWrapper = withFormik<FormProps & EditFormProps, FormValues>({
  enableReinitialize: true,
  mapPropsToValues: ({ initialValues }) => {
    let { date } = initialValues;
    if (date) {
      date = moment(date).format(moment.HTML5_FMT.DATETIME_LOCAL);
    }
    return { ...initialValues, date };
  },
  handleSubmit: async (values, { setStatus, props: { onSubmit } }) => {
    if (values.date === "") {
      values = omit(values, "date") as FormValues;
    }
    try {
      await onSubmit(values);
    } catch {
      setStatus("Something went weird.");
    }
  }
})(EditForm);

export class Upsert extends Component<Props> {
  static defaultProps = {
    tags: []
  };

  componentDidMount() {
    this.props.fetchItem(this.props.id);
  }

  handleSubmit = async (values: any) => {
    await this.props.upsertItem({
      ...values,
      amount: Number(values.amount),
      id: this.props.id
    });
    this.props.history.push("/");
  };

  handleDelete = async () => {
    try {
      await this.props.deleteItem(this.props.id);
      this.props.history.push("/");
    } catch (errors) {
      console.log("something went weird", errors); // eslint-disable-line no-console
    }
  };

  render() {
    console.log(this.props);
    const { id, tags, initialValues, isSubmitting } = this.props;
    return (
      <Container>
        <Title>{id ? "Edit" : "Add"}</Title>
        <EditWrapper
          onSubmit={this.handleSubmit}
          tags={tags}
          showDelete={id ? true : false}
          deleteCallback={this.handleDelete}
          initialValues={initialValues}
          isSubmitting={isSubmitting}
        />
      </Container>
    );
  }
}

const mapStateToProps = (state: any, ownProps: any) => {
  const defaultInitialValues: FormValues = {
    date: "",
    amount: "",
    description: "",
    tags: []
  };
  const {
    item: {
      user: { item: initialValues = defaultInitialValues, tags },
      errors
    }
  } = state;
  const {
    match: {
      params: { id = undefined }
    }
  } = ownProps;

  const existingTags =
    initialValues.tags && initialValues.tags.edges
      ? initialValues.tags.edges.map((tag: any) => tag.node)
      : [];

  const availableTags =
    tags && tags.edges ? tags.edges.map((tag: any) => tag.node) : [];

  return {
    ...state,
    tags: availableTags,
    initialValues: { ...initialValues, tags: existingTags },
    errors,
    id
  };
};

export default connect(mapStateToProps, {
  upsertItem,
  fetchItem,
  deleteItem
})(Upsert);
