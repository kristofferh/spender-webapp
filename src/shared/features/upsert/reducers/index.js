import {
  UPSERT_ITEM_REQUEST,
  UPSERT_ITEM_SUCCESS,
  UPSERT_ITEM_FAILURE,
  FETCH_ITEM_REQUEST,
  FETCH_ITEM_SUCCESS,
  FETCH_ITEM_FAILURE
} from "../constants";

export default (
  state = {
    isFetching: false,
    isSubmitting: false,
    item: {},
    errors: {}
  },
  action
) => {
  switch (action.type) {
    case UPSERT_ITEM_REQUEST:
      return {
        ...state,
        isSubmitting: true
      };
    case UPSERT_ITEM_SUCCESS:
      return {
        ...state,
        isSubmitting: false,
        item: action.item
      };
    case UPSERT_ITEM_FAILURE:
      return {
        ...state,
        isSubmitting: false,
        errors: action.errors
      };
    case FETCH_ITEM_REQUEST:
      return {
        ...state,
        isFetching: true
      };
    case FETCH_ITEM_SUCCESS:
      return {
        ...state,
        isFetching: false,
        item: action.item
      };
    case FETCH_ITEM_FAILURE:
      return {
        ...state,
        isFetching: false,
        errors: action.errors
      };
    default:
      return state;
  }
};
