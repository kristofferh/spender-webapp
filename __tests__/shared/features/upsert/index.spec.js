import React from "react";
import { reduxForm } from "redux-form";
import { createStore } from "redux";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import "jest-emotion";

import { Upsert } from "shared/features/upsert";

const spy = jest.fn();
const store = createStore(() => ({}));

const Decorated = reduxForm({ form: "testForm" })(Upsert);

describe("Create snapshot", () => {
  it("should render and match the snapshot for adding an item", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Decorated onSubmit={spy} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Create snapshot", () => {
  it("should render and match the snapshot for editing an item", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Decorated
            onSubmit={spy}
            initialValues={{ date: "July 12, 2008", amount: 10 }}
            id={5}
            fetchItem={spy}
          />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
