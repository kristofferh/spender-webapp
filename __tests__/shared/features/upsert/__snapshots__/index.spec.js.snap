// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create snapshot should render and match the snapshot for adding an item 1`] = `
.emotion-52 {
  padding: 0 15px;
}

@media (min-width:768px) {
  .emotion-52 {
    max-width: 960px;
    margin: 30px auto;
  }
}

@media (max-width:767px) {
  .emotion-52 {
    padding: 15px;
  }
}

.emotion-0 {
  font-family: 'tiempos-head-semibold',georgia,serif;
  font-size: 28px;
  line-height: 1.2;
  margin-top: 0;
}

.emotion-50 {
  padding: 0 15px;
  margin: 0;
  background: #ffffff;
}

@media (min-width:768px) {
  .emotion-50 {
    max-width: 960px;
    margin: 30px auto;
  }
}

@media (min-width:768px) {
  .emotion-50 {
    background: #ffffff;
    box-shadow: 0 2px 5px rgba(17,17,17,0.1);
    border-radius: 2px;
    padding: 20px 30px;
  }
}

@media (max-width:767px) {
  .emotion-50 {
    margin: 0 -15px;
    padding: 15px;
  }
}

.emotion-10 {
  display: block;
  margin-bottom: 15px;
}

.emotion-8 {
  position: relative;
}

.emotion-2 {
  font-weight: bold;
  display: block;
}

.emotion-6 {
  width: 100%;
  border: 0;
  border-bottom: 1px solid #888;
  padding: 5px 0;
  -webkit-transition: all 2s ease;
  transition: all 2s ease;
}

.emotion-6:focus {
  outline: none;
  border-image: linear-gradient(156deg,#e1cad3 6%,#a6a8bc 96%) 2;
}

.emotion-26 {
  width: 100%;
  border: 0;
  border-bottom: 1px solid #888;
  padding: 5px 0;
  -webkit-transition: all 2s ease;
  transition: all 2s ease;
  height: 30px;
  resize: none;
}

.emotion-26:focus {
  outline: none;
  border-image: linear-gradient(156deg,#e1cad3 6%,#a6a8bc 96%) 2;
}

.emotion-45 {
  position: relative;
  box-sizing: border-box;
}

.emotion-36 {
  border: 0;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-44 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: hsl(0,0%,98%);
  border-color: hsl(0,0%,80%);
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  cursor: default;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-height: 38px;
  outline: 0 !important;
  position: relative;
  -webkit-transition: all 100ms;
  transition: all 100ms;
  box-sizing: border-box;
}

.emotion-44:hover {
  border-color: hsl(0,0%,70%);
}

.emotion-39 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 2px 8px;
  -webkit-overflow-scrolling: touch;
  position: relative;
  box-sizing: border-box;
}

.emotion-37 {
  color: hsl(0,0%,50%);
  margin-left: 2px;
  margin-right: 2px;
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  box-sizing: border-box;
}

.emotion-38 {
  margin: 2px;
  padding-bottom: 2px;
  padding-top: 2px;
  visibility: visible;
  color: #222;
  box-sizing: border-box;
}

.emotion-43 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
}

.emotion-40 {
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  background-color: hsl(0,0%,80%);
  margin-bottom: 8px;
  margin-top: 8px;
  width: 1px;
  box-sizing: border-box;
}

.emotion-42 {
  color: hsl(0,0%,80%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px;
  -webkit-transition: color 150ms;
  transition: color 150ms;
  box-sizing: border-box;
}

.emotion-42:hover {
  color: hsl(0,0%,60%);
}

.emotion-41 {
  display: inline-block;
  fill: currentColor;
  line-height: 1;
  stroke: currentColor;
  stroke-width: 0;
}

<div
  className="emotion-52 emotion-53"
>
  <h1
    className="emotion-0 emotion-1"
  >
    Add
  </h1>
  <form
    className="emotion-50 emotion-51"
    onSubmit={[Function]}
  >
    <div
      className="emotion-10 emotion-11"
    >
      <div
        className="form-group emotion-8 emotion-9"
      >
        <div
          className="emotion-4 emotion-5"
        >
          <label
            className="emotion-2 emotion-3"
            htmlFor="date"
            required={false}
          >
            Date
          </label>
        </div>
        <input
          className="form-control emotion-6 emotion-7"
          id="date"
          name="date"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          type="datetime-local"
          value=""
        />
      </div>
    </div>
    <div
      className="emotion-10 emotion-11"
    >
      <div
        className="form-group emotion-8 emotion-9"
      >
        <div
          className="emotion-4 emotion-5"
        >
          <label
            className="emotion-2 emotion-3"
            htmlFor="amount"
            required={false}
          >
            Amount
          </label>
        </div>
        <input
          className="form-control emotion-6 emotion-7"
          id="amount"
          name="amount"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          step="any"
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="emotion-10 emotion-11"
    >
      <div
        className="form-group emotion-8 emotion-9"
      >
        <div
          className="emotion-4 emotion-5"
        >
          <label
            className="emotion-2 emotion-3"
            htmlFor="description"
            required={false}
          >
            Description
          </label>
        </div>
        <textarea
          className="form-control emotion-26 emotion-27"
          id="description"
          name="description"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="emotion-10 emotion-11"
    >
      <div
        className="form-group emotion-8 emotion-9"
      >
        <div
          className="emotion-4 emotion-5"
        >
          <label
            className="emotion-2 emotion-3"
            htmlFor="tags"
            required={false}
          />
        </div>
        <div
          className="emotion-45"
          id="tags"
          onKeyDown={[Function]}
        >
          <span
            aria-atomic="true"
            aria-live="polite"
            className="emotion-36"
            role="status"
          >
            0 results available.
          </span>
          <div
            className="emotion-44"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className="emotion-39"
            >
              <div
                className="emotion-37"
              >
                Select...
              </div>
              <div
                className="emotion-38"
              >
                <div
                  className=""
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-autocomplete="list"
                    aria-busy={false}
                    aria-expanded={false}
                    aria-haspopup={false}
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    disabled={false}
                    id="react-select-2-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    role="combobox"
                    spellCheck="false"
                    style={
                      Object {
                        "background": 0,
                        "border": 0,
                        "boxSizing": "content-box",
                        "color": "inherit",
                        "fontSize": "inherit",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "1px",
                      }
                    }
                    tabIndex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </div>
            </div>
            <div
              className="emotion-43"
            >
              <span
                className="emotion-40"
                role="presentation"
              />
              <div
                className="emotion-42"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
                role="button"
              >
                <svg
                  className="emotion-41"
                  focusable="false"
                  height={20}
                  role="presentation"
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div />
        </div>
      </div>
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`Create snapshot should render and match the snapshot for editing an item 1`] = `
.emotion-54 {
  padding: 0 15px;
}

@media (min-width:768px) {
  .emotion-54 {
    max-width: 960px;
    margin: 30px auto;
  }
}

@media (max-width:767px) {
  .emotion-54 {
    padding: 15px;
  }
}

.emotion-0 {
  font-family: 'tiempos-head-semibold',georgia,serif;
  font-size: 28px;
  line-height: 1.2;
  margin-top: 0;
}

.emotion-52 {
  padding: 0 15px;
  margin: 0;
  background: #ffffff;
}

@media (min-width:768px) {
  .emotion-52 {
    max-width: 960px;
    margin: 30px auto;
  }
}

@media (min-width:768px) {
  .emotion-52 {
    background: #ffffff;
    box-shadow: 0 2px 5px rgba(17,17,17,0.1);
    border-radius: 2px;
    padding: 20px 30px;
  }
}

@media (max-width:767px) {
  .emotion-52 {
    margin: 0 -15px;
    padding: 15px;
  }
}

.emotion-10 {
  display: block;
  margin-bottom: 15px;
}

.emotion-8 {
  position: relative;
}

.emotion-2 {
  font-weight: bold;
  display: block;
}

.emotion-6 {
  width: 100%;
  border: 0;
  border-bottom: 1px solid #888;
  padding: 5px 0;
  -webkit-transition: all 2s ease;
  transition: all 2s ease;
}

.emotion-6:focus {
  outline: none;
  border-image: linear-gradient(156deg,#e1cad3 6%,#a6a8bc 96%) 2;
}

.emotion-26 {
  width: 100%;
  border: 0;
  border-bottom: 1px solid #888;
  padding: 5px 0;
  -webkit-transition: all 2s ease;
  transition: all 2s ease;
  height: 30px;
  resize: none;
}

.emotion-26:focus {
  outline: none;
  border-image: linear-gradient(156deg,#e1cad3 6%,#a6a8bc 96%) 2;
}

.emotion-45 {
  position: relative;
  box-sizing: border-box;
}

.emotion-36 {
  border: 0;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-44 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: hsl(0,0%,98%);
  border-color: hsl(0,0%,80%);
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  cursor: default;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-height: 38px;
  outline: 0 !important;
  position: relative;
  -webkit-transition: all 100ms;
  transition: all 100ms;
  box-sizing: border-box;
}

.emotion-44:hover {
  border-color: hsl(0,0%,70%);
}

.emotion-39 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 2px 8px;
  -webkit-overflow-scrolling: touch;
  position: relative;
  box-sizing: border-box;
}

.emotion-37 {
  color: hsl(0,0%,50%);
  margin-left: 2px;
  margin-right: 2px;
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  box-sizing: border-box;
}

.emotion-38 {
  margin: 2px;
  padding-bottom: 2px;
  padding-top: 2px;
  visibility: visible;
  color: #222;
  box-sizing: border-box;
}

.emotion-43 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
}

.emotion-40 {
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  background-color: hsl(0,0%,80%);
  margin-bottom: 8px;
  margin-top: 8px;
  width: 1px;
  box-sizing: border-box;
}

.emotion-42 {
  color: hsl(0,0%,80%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 8px;
  -webkit-transition: color 150ms;
  transition: color 150ms;
  box-sizing: border-box;
}

.emotion-42:hover {
  color: hsl(0,0%,60%);
}

.emotion-41 {
  display: inline-block;
  fill: currentColor;
  line-height: 1;
  stroke: currentColor;
  stroke-width: 0;
}

.emotion-50 {
  cursor: pointer;
  display: block;
  margin-top: 15px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
  className="emotion-54 emotion-55"
>
  <h1
    className="emotion-0 emotion-1"
  >
    Edit
  </h1>
  <form
    className="emotion-52 emotion-53"
    onSubmit={[Function]}
  >
    <div
      className="emotion-10 emotion-11"
    >
      <div
        className="form-group emotion-8 emotion-9"
      >
        <div
          className="emotion-4 emotion-5"
        >
          <label
            className="emotion-2 emotion-3"
            htmlFor="date"
            required={false}
          >
            Date
          </label>
        </div>
        <input
          className="form-control emotion-6 emotion-7"
          id="date"
          name="date"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          type="datetime-local"
          value=""
        />
      </div>
    </div>
    <div
      className="emotion-10 emotion-11"
    >
      <div
        className="form-group emotion-8 emotion-9"
      >
        <div
          className="emotion-4 emotion-5"
        >
          <label
            className="emotion-2 emotion-3"
            htmlFor="amount"
            required={false}
          >
            Amount
          </label>
        </div>
        <input
          className="form-control emotion-6 emotion-7"
          id="amount"
          name="amount"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          step="any"
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="emotion-10 emotion-11"
    >
      <div
        className="form-group emotion-8 emotion-9"
      >
        <div
          className="emotion-4 emotion-5"
        >
          <label
            className="emotion-2 emotion-3"
            htmlFor="description"
            required={false}
          >
            Description
          </label>
        </div>
        <textarea
          className="form-control emotion-26 emotion-27"
          id="description"
          name="description"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="emotion-10 emotion-11"
    >
      <div
        className="form-group emotion-8 emotion-9"
      >
        <div
          className="emotion-4 emotion-5"
        >
          <label
            className="emotion-2 emotion-3"
            htmlFor="tags"
            required={false}
          />
        </div>
        <div
          className="emotion-45"
          id="tags"
          onKeyDown={[Function]}
        >
          <span
            aria-atomic="true"
            aria-live="polite"
            className="emotion-36"
            role="status"
          >
            0 results available.
          </span>
          <div
            className="emotion-44"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className="emotion-39"
            >
              <div
                className="emotion-37"
              >
                Select...
              </div>
              <div
                className="emotion-38"
              >
                <div
                  className=""
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-autocomplete="list"
                    aria-busy={false}
                    aria-expanded={false}
                    aria-haspopup={false}
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    disabled={false}
                    id="react-select-3-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    role="combobox"
                    spellCheck="false"
                    style={
                      Object {
                        "background": 0,
                        "border": 0,
                        "boxSizing": "content-box",
                        "color": "inherit",
                        "fontSize": "inherit",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "1px",
                      }
                    }
                    tabIndex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </div>
            </div>
            <div
              className="emotion-43"
            >
              <span
                className="emotion-40"
                role="presentation"
              />
              <div
                className="emotion-42"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
                role="button"
              >
                <svg
                  className="emotion-41"
                  focusable="false"
                  height={20}
                  role="presentation"
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div />
        </div>
      </div>
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
    <a
      className="emotion-50 emotion-51"
      onClick={[Function]}
    >
      Delete
    </a>
  </form>
</div>
`;
